scapy/automaton.py

```diff
@@ -1088,8 +1100,17 @@ def _do_control(self, ready, *args, **kargs):
 
             # Start the automaton
             self.state = self.initial_states[0](self)
-            self.send_sock = self.send_sock_class(**self.socket_kargs)
-            self.listen_sock = self.recv_sock_class(**self.socket_kargs)
+            if "iface" in self.socket_kargs and \
+               isinstance(self.socket_kargs["iface"], list):
+                kargs = dict(self.socket_kargs)
+                del kargs["iface"]
```

Comment:
this is kinda blury and could be cleaned up a bit. I'd prefer having a single `self.send_socks = ` and `self.listen_socks =`, with eventually a list containing only one element in the case `iface` isn't used, or isn't a list.

Reply:
Would something like this be better?
```python
if "iface" in self.socket_kargs and \
   isinstance(self.socket_kargs["iface"], list):
    kwargslist = [dict(self.socket_kargs) for _ in self.socket_kargs["iface"]]
    for dct, ifs in zip(kwargslist, self.socket_kargs["iface"]):
	dct["iface"] = ifs
else:
    kwargslist = [self.socket_kargs]
self.send_socks = [self.send_sock_class(**kargs) for kwargs in kwargslist]
self.listen_socks = [self.recv_sock_class(**kargs) for kwargs in kwargslist]
```